src/saltext/vmware/modules/esxi.py

```diff
@@ -845,6 +845,326 @@ def get_ntp_config(
         raise salt.exceptions.SaltException(str(exc))
 
 
+def list_hosts(
+    datacenter_name=None,
+    cluster_name=None,
+    host_name=None,
+    service_instance=None,
+):
+    """
+    List ESXi hosts.
+
+    datacenter_name
+        Filter by this datacenter name (required when cluster is specified)
+
+    cluster_name
+        Filter by this cluster name (optional)
+
+    host_name
+        Filter by this ESXi hostname (optional)
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    .. code-block:: bash
+
+        salt '*' vmware_esxi.list_hosts
+    """
+    log.debug("Running vmware_esxi.list_hosts")
+    ret = []
+    if not service_instance:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+    hosts = utils_esxi.get_hosts(
+        service_instance=service_instance,
+        host_names=[host_name] if host_name else None,
+        cluster_name=cluster_name,
+        datacenter_name=datacenter_name,
+        get_all_hosts=host_name is None,
+    )
+
+    try:
+        for h in hosts:
+            ret.append(h.name)
+        return ret
+    except (
+        vim.fault.InvalidState,
+        vim.fault.NotFound,
+        vim.fault.HostConfigFault,
+        vmodl.fault.InvalidArgument,
+        salt.exceptions.VMwareApiError,
+        vim.fault.AlreadyExists,
+    ) as exc:
+        raise salt.exceptions.SaltException(str(exc))
+
+
+def add_user(
```

Comment:
Should `add_user` and `update_user` also accept privileges that should be enabled?

I can do it, is there a reason you didn't add those variables? 

(I'm thinking that maybe it's not done because we don't have code to manage privileges yet?)

Reply:
The ticket + module here (https://github.com/ansible-collections/community.vmware/blob/main/plugins/modules/vmware_local_user_manager.py) didn't require me to add them, so just went with the ticket. You can add them if you have a ticket to that effect.

I have another ticket with role that deals with permissions. There is also a group + associating user to a group. But, as Joey said, we are going by priority and dealing with the cluster functionality w/ individual tickets later. (suboptimal but that is how we are doing it now)

Reply:
I'm confused because reading 3230 I see "I want to add a local role with privileges to an ESXi host".

I didn't see another ticket that deals with privileges, I'm glad you know where it is :)

Now I'm confused between VRAE-3230 and VRAE-3235, one is for roles the other for users? So I guess this PR is for 3235?

Reply:
I am so stupid. I confused you. This PR is for 3235.

Reply:
Sorry

Reply:
We all know you're not stupid, and we all make mistakes :)

Thanks for the clarification!